keras_nlp/src/models/deeplab_v3/deeplab_v3_backbone.py

```diff
@@ -0,0 +1,235 @@
+        property such as  "P4", "P5" etc.
+    upsampling_size: Int, or tuple of 2 integers. The upsampling factors for
+        rows and columns of `spatial_pyramid_pooling` layer.
+        If `low_level_feature_key` is given then `spatial_pyramid_pooling`s
+        layer resolution should match with the `low_level_feature`s layer
+        resolution to concatenate both the layers for combined encoder outputs.
+    dilation_rates: A `list` of integers for parallel dilated conv.
+        Applied only when Default `SpatialPyramidPooling` is used. Usually a
+        sample choice of rates are [6, 12, 18].
+    low_level_feature_key: (Optional) str, layer level to extract the feature
+        from one of the key from the `backbone`s `pyramid_outputs`
+        property such as  "P2", "P3" etc which will be the Decoder block.
+        Required only when the DeepLabV3Plus architecture needs to be applied.
+    activation: str or callable. The activation function to use on
+        the `Dense` layer. Set `activation=None` to return the output
+        logits. Defaults to `"softmax"`.
+    spatial_pyramid_pooling: (Optional) a `keras.layers.Layer`. Also known
+        as Atrous Spatial Pyramid Pooling (ASPP). Performs spatial pooling
+        on different spatial levels in the pyramid, with dilation. If
+        provided, the feature map from the backbone is passed to it inside
+        the DeepLabV3 Encoder, otherwise SpatialPyramidPooling layer is used.
+    segmentation_head: (Optional) a `keras.layers.Layer`. If provided, the
+        outputs of the DeepLabV3 encoder is passed to this layer and it
+        will be considered as the last layer before final segmentaion layer ,
+        otherwise a default DeepLabV3 convolutional head is used.
+
+    Example:
+    ```python
+    image_encoder = keras_nlp.models.ResNetBackbone.from_preset("resnet_v2_50")
+
+    model = keras_nlp.models.DeepLabV3Backbone(
+        image_encoder= image_encoder,
+        projection_filters=48,
+        low_level_feature_key="P2",
+        spatial_pyramid_pooling_key="P5",
+        upsampling_size = 8,
+        dilation_rates = [6, 12, 18]
+    )
+    ```
+    """
+
+    def __init__(
+        self,
+        image_encoder,
+        spatial_pyramid_pooling_key,
+        upsampling_size,
+        dilation_rates,
+        low_level_feature_key=None,
+        projection_filters=48,
+        spatial_pyramid_pooling=None,
+        segmentation_head=None,
+        **kwargs,
+    ):
+        if not isinstance(image_encoder, keras.Model):
+            raise ValueError(
+                "Argument `image_encoder` must be a `keras.Model` instance. Received instead "
+                f"backbone={image_encoder} (of type {type(image_encoder)})."
+            )
+        data_format = keras.config.image_data_format()
+        channel_axis = -1 if data_format == "channels_last" else 1
+        # === Functional Model ===
+        inputs = keras.layers.Input((None, None, 3))
```

Comment:
I think we are taking in `image_size` generally, and defaulting it `(None, None, 3)`, primarily so that users can pass a different number of channels.